R/calc_variables_stem_level.R

```diff
@@ -0,0 +1,254 @@
+              .data$f * .data$calc_height_m * .data$perimeter +
+              .data$g * .data$calc_height_m * .data$perimeter ^ 2,
+            no =
+              1 / 1000 *
+              #spil
+              (exp(1.10597 * log(.data$calc_height_m) + 1.78865 * log(.data$d_cm) - 3.07192) -
+                 #Verlies
+                 exp(-4.608923 * log(.data$d_cm) + 3.005989 * log(.data$calc_height_m) -
+                       1.3209 * log(.data$calc_height_m) * log(.data$calc_height_m) +
+                       1.605266 * log(.data$d_cm) * log(.data$calc_height_m) + 5.410272))
+          )
+        , vol_bole_t2_m3 = pmax(0, .data$vol_bole_t2_m3)
+        , vol_bole_m3 =
+          ifelse(
+            .data$ind_sht_cop == 12 & is.na(.data$vol_bole_t2_m3),
+            .data$vol_bole_t1_m3,
+            .data$vol_bole_t2_m3
+          )
+      ) %>%
+    select(
+      -.data$a, -.data$b, -.data$c, -.data$d, -.data$e, -.data$f, -.data$g,
+      -.data$formula, -.data$d_cm, -.data$perimeter
+    ) %>%
+    mutate(
+      # volume correction for snags
+      vol_crown_m3 = ifelse(.data$intact_snag == 10, 0, .data$vol_crown_m3),
+      upper_diam_snag_mm = ifelse(.data$intact_snag == 10,
+                                    .data$dbh_mm * (.data$calc_height_m - .data$height_m) / .data$calc_height_m,
+                                    NA),
+      volume_snag_m3 = ifelse(.data$intact_snag == 10,
+                              # as truncated cone - appears to be less accurate!!!!
+                              # pi * .data$height_m * (.data$dbh_mm^2 + .data$dbh_mm * .data$upper_diam_snag_mm + .data$upper_diam_snag_mm^2) / (3 * 2000^2),
+                              # 1/3 x π x h x ( R² + R x r + r² ) - truncated cone
+                              # AS CILINDER - GIVES BETTER RESULTS
+                              pi * .data$height_m * .data$dbh_mm^2 / 2000^2,
+                              # ! TEMPORARY SOLUTION: goal is to incormporate taper functuions cfr FM-IA
+                                NA),
+      # !!! ? als calc_height er niet is, dan ev. wel nog als cilinder???
+      # nee, want dan ook geen volumes van de andere bomen ...)
+      vol_bole_m3 = ifelse(.data$intact_snag == 10,
+                             .data$volume_snag_m3,
+                             .data$vol_bole_m3)
+    ) %>%
+    mutate(
+      # volume correction for broken crown or branches
+      reduction_crown =
+        ifelse(is.na(.data$crown_volume_reduction), 0, .data$crown_volume_reduction),
+      vol_crown_m3 = .data$vol_crown_m3 * (1 - .data$reduction_crown),
+      reduction_branch =
+        ifelse(is.na(.data$branch_length_reduction), 0, .data$branch_length_reduction),
+      vol_crown_m3 = .data$vol_crown_m3 * (1 - .data$reduction_branch),
+      # total volume
+      vol_tot_m3 = .data$vol_bole_m3 + .data$vol_crown_m3
+    ) %>%
+    select(-upper_diam_snag_mm, -volume_snag_m3)
```

Comment:
```suggestion
    select(-.data$upper_diam_snag_mm, -.data$volume_snag_m3)
```
Toch nog hier en daar .data's vergeten.  (Je krijgt die netjes opgelijst in een note als je op tab `Build` de `Check` runt

Reply:
ja, er zijn er nog, ik pas die aan in mijn scripts!

Reply:
ik krijg nog één "note" als ik op check klik, maar weet niet wat die betekent:
> checking top-level files ... NOTE
  Non-standard file/directory found at top level:
    'data-raw'

Reply:
Blijkbaar mag er (volgens de stijl die CRAN eist voor packages die daarop gepubliceerd worden) geen folder met naam `data-raw` in de hoofdfolder van het project zitten.  Mogelijk zien ze dit niet graag omdat die 'manueel onderhoud' betekent?  Zelf heb ik dit idee (folder `data-raw` met hierin scripts om bepaalde data te genereren) van een suggestie van het internet en van een ander package overgenomen, maar dat gaat dan ook niet op CRAN staan...

In elk geval: zolang we niet de bedoeling hebben om ons package op CRAN te publiceren, zijn we niet verplicht om aan al die regeltjes te voldoen als we ze niet relevant vinden voor onze situatie.  We proberen uiteraard zo goed mogelijk te doen (dit garandeert ook dat anderen weinig problemen ondervinden als ze het willen installeren, dus het is in ons eigen voordeel), maar we kunnen er ook voor kiezen om een bepaalde note te negeren omdat we ze minder relevant vinden.  (Ook het package `checklist` dat Thierry gemaakt heeft om automatisch dit soort controles toe te voegen op INBO-packages, laat trouwens toe om welbepaalde notes te negeren.)

Ah, misschien de gemakkelijkste manier om deze note op te lossen: deze folder toevoegen aan het bestand `.Rbuildignore`.